tools/src/tokenize.rs

```diff
@@ -266,7 +266,77 @@ pub fn tokenize_c_like(string: &String, spec: &LanguageSpec) -> Vec<Token> {
                 tokens.push(Token {start: start, end: peek_pos(), kind: TokenKind::StringLiteral});
             }
             next_token_maybe_regexp_literal = true;
-        } else if ch == '\'' || ch == '"' {
+        } else if ch == 'R' {
+            // Handle raw literals.
+            assert!(peek_char() == '"');
+            get_char();
+
+            // Read delimiter.
+            loop {
+                let c = peek_char();
+                if c == '(' {
+                    break;
+                }
+
+                if c == ')' || c == '\\' || is_whitespace(c) {
+                    writeln!(&mut std::io::stderr(), "Illegal character in raw string delimiter").unwrap();
+                    return tokens;
+                }
+
+                get_char();
+
+                // NB: Start points to the R" so subtract 2 to find the length
+                // of the sentinel.
+                if peek_pos() - start - 2 > 16 {
+                    writeln!(&mut std::io::stderr(), "Delimiter for raw string longer than 16 characters").unwrap();
+                    return tokens;
+                }
+
+                if peek_pos() == string.len() {
+                    break;
+                }
+            }
+
+            if peek_pos() == string.len() || peek_char() != '(' {
+                writeln!(&mut std::io::stderr(), "Expecting '(' after raw string literal").unwrap();
+                return tokens;
+            }
+
+            let sentinel = &string[start + 2 .. cur_pos.get()];
```

Comment:
I think you want to use peek_pos() instead of cur_pos.get(). cur_pos gives you the index into the chars vector. If the string contains multibyte characters, then chars and string won't line up.

Or you could use the result of get_char() as I described above.